dcap/quoteverify/src/verify.rs

```diff
@@ -2,8 +2,43 @@
 
 //! This module contains logic to verify a DCAP quote
 
+use mc_sgx_core_types::{BaseQuote, Quote};
+use mc_sgx_dcap_sys_types::{sgx_ql_qv_result_t, time_t};
 use mc_sgx_dcap_types::Quote3Error;
 use mc_sgx_util::ResultInto;
+use std::{
+    ptr,
+    time::{SystemTime, UNIX_EPOCH},
+};
+
+pub trait Verify: BaseQuote {
+    fn verify(&self) -> Result<(), Quote3Error> {
+        let raw_quote = self.raw_quote();
+        let start = SystemTime::now();
+        let since_the_epoch = start
+            .duration_since(UNIX_EPOCH)
+            .expect("Time went backwards");
+        let mut expiration_status = 0;
+        let mut quote_status = sgx_ql_qv_result_t::SGX_QL_QV_RESULT_MAX;
+
+        unsafe {
+            mc_sgx_dcap_quoteverify_sys::sgx_qv_verify_quote(
+                raw_quote.as_ref().as_ptr(),
+                raw_quote.as_ref().len() as u32,
+                ptr::null(),
+                since_the_epoch.as_secs() as time_t,
+                &mut expiration_status,
+                &mut quote_status,
```

Comment:
up for suggestions on how to best chain these results and how best to prioritize which return value should win, or if we should pass some kind of error struct

Reply:
I don't really understand what you mean by "which return value should win"?